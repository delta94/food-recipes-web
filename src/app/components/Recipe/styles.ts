import styled from 'styled-components'

export const RecipeContainer = styled.article`
    display: flex;
    max-width: 45%;
    margin: 1rem;
    background-color: white;
    border-radius: 16px;
    /* flex-wrap: wrap; */

    @media screen and (max-width: 1600px) {
        max-width: 45%;
    }
    @media screen and (max-width: 950px) {
        max-width: 100%;
    }

    @media screen and (max-width: 600px) {
        display: block;
    }
`

export const RecipeLeft = styled.div`
    img {
        object-fit: cover;
        object-position: center;
        max-width: 250px;
        height: 300px;
        border-top-left-radius: 16px;
        border-bottom-left-radius: 16px;
    }

    @media screen and (max-width: 600px) {
        img {
            max-width: 100%;
            height: auto;
            border-radius: 16px;
        }
    }
`

export const RecipeRight = styled.div`

    display: flex;
    padding-left: 1rem;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;

    h2, p {
        /* margin: 0.9rem;
        text-align: center; */
    }

    @media screen and (max-width: 600px) {
        display: block;

        h2, p {
            margin: 1rem;
        }
    }

`

export const RecipeTime = styled.div`
    span {
        margin: 1rem;
        font-weight: bold;
    }
`
