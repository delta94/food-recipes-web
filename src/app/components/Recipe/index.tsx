import React from 'react'
import { Link } from 'react-router-dom'

import { RecipeContainer, RecipeLeft, RecipeRight, RecipeTime } from './styles'

const RecipeM = () => {
  return (
    <RecipeContainer>
      <RecipeLeft>
        <Link to="/recipes/recipe">
          <img
            src='https://images.unsplash.com/photo-1490818387583-1baba5e638af?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1231&q=80'
            alt='Food'
          />
        </Link>
      </RecipeLeft>
      <RecipeRight>
        <h2>Receita da vov√≥</h2>
        <p>
          Lorem Ipsum is simply dummy text of the printing and typesetting
          industry. Lorem Ipsum has been the industry's standard dummy text ever
          since the 1500s
        </p>
        <RecipeTime>
          <span>Tempo de preparo:</span>
          <p>1h</p>
        </RecipeTime>
      </RecipeRight>
    </RecipeContainer>
  )
}

export default RecipeM
