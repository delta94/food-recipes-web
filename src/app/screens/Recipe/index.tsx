import React from 'react'
import { Link } from 'react-router-dom'
import logo from '../../../assets/img/logo.png'

import { Container, ContainerButtons } from './styles'

const Recipe = () => {
  return (
    <Container>
      <h1>Recipe</h1>
      <img
        src='https://images.unsplash.com/photo-1490818387583-1baba5e638af?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1231&q=80'
        alt=''
      />

      <h2>Ingredientes</h2>
      <p>
        2 pacotes de bolo pronto de chocolate ( tipo Pulmann) 1 copo de guaraná
        2 latas de leite condensado 100 g de coco ralado 2 colheres (sopa) de
        margarina 200 g de chocolate meio amargo 1 lata de creme de leite com
        soro coco ralado e raspas de chocolate para decorar
      </p>

      <h2>Modo de preparo</h2>
      <p>
        Leve ao fogo as 2 latas de leite condensado com a margarina e o coco
        ralado até formar um beijinho em ponto mole. Reserve. Pique o chocolate
        e leve ao micro-ondas por 45 segundos. Adicione o creme de leite e mexa
        bem até formar um creme brilhante, a ganache.
      </p>

      <h2>Tempo</h2>
      <p>1h 15min</p>

      <ContainerButtons>
        <button type='submit'>Editaraaaaaaaaaaa</button>
        <button type='submit'>Apagar</button>
      </ContainerButtons>
    </Container>

  )
}

export default Recipe
