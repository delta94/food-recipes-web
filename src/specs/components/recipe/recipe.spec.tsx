import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Recipe from '../../../app/components/Recipe'

import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'

const mockStore = configureStore([])

describe('[COMPONENTS] Recipe', () => {
  let store: any

  beforeEach(() => {
    store = mockStore({
      auth: {}
    })
  })

  it('Should render correctly', () => {
    const container = render(
      <BrowserRouter>
        <Provider store={store}>
          <Recipe
            id='any_id'
            name='any_name'
            image='any_image'
            ingredients='any_ingredients'
            mode_prepare='any_mode_prepare'
            time='any_time'
          />
        </Provider>
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })

  it('Should call function openModal on click modal', async () => {
    const { getByRole } = render(
      <BrowserRouter>
        <Provider store={store}>
          <Recipe
            id='any_id'
            name='any_name'
            image='any_image'
            ingredients='any_ingredients'
            mode_prepare='any_mode_prepare'
            time='any_time'
          />
        </Provider>
      </BrowserRouter>
    )

    expect(fireEvent.click(getByRole('img'))).toBeTruthy()
  })

  // it('Should call function closeModal on click modal', async () => {
  //   const { getByRole } = render(
  //     <BrowserRouter>
  //       <Provider store={store}>
  //         <Recipe
  //           id='any_id'
  //           name='any_name'
  //           image='any_image'
  //           ingredients='any_ingredients'
  //           mode_prepare='any_mode_prepare'
  //           time='any_time'
  //         />
  //       </Provider>
  //     </BrowserRouter>
  //   )

  //   fireEvent.click(getByRole('img'))
  //   expect(fireEvent.click(document.body)).toBeTruthy()
  // })
})
