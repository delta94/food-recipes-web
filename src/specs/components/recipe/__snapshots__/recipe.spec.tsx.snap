// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] Recipe Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="sc-dlnjPT clImnd"
      >
        <div
          class="sc-hKFyIo kTaKEQ"
        >
          <img
            alt="any_name"
            src="http://localhost:3333/files/any_image"
          />
        </div>
        <div
          class="sc-eCApGN dDnMsO"
        >
          <h2>
            any_name
          </h2>
          <p>
            any_mode_prepare
          </p>
          <div
            class="sc-jSFkmK epcJIx"
          >
            <span>
              Tempo de preparo:
            </span>
            <p>
              any_time
            </p>
          </div>
        </div>
      </article>
    </div>
    <div
      class="ReactModalPortal"
    />
  </body>,
  "container": <div>
    <article
      class="sc-dlnjPT clImnd"
    >
      <div
        class="sc-hKFyIo kTaKEQ"
      >
        <img
          alt="any_name"
          src="http://localhost:3333/files/any_image"
        />
      </div>
      <div
        class="sc-eCApGN dDnMsO"
      >
        <h2>
          any_name
        </h2>
        <p>
          any_mode_prepare
        </p>
        <div
          class="sc-jSFkmK epcJIx"
        >
          <span>
            Tempo de preparo:
          </span>
          <p>
            any_time
          </p>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
